skupstina_django/scraper/scrape_utils.py

```diff
@@ -0,0 +1,127 @@
+import re
+import requests
+from bs4 import BeautifulSoup
+from skupstina.models import Category, Source, Item, Act
+
+root_url = 'http://web.zagreb.hr'
+
+
+def parse_subjects_list(url):
+    site = requests.get(url).content
+    soup = BeautifulSoup(site, 'html.parser')
+    els = soup.select('.centralTD ul ol li')
+    print(len(els), ' elements')
+    subjects = []
+    for el in els:
+        subel = el.contents[0]
+        print(subel)
+        try:
+            link = subel.attrs['href']
+        except AttributeError:
+            continue
+        title = subel.select('b')[0].contents[0]
+        subjects.append({'title:': title, 'subject_url': root_url + link})
+    return subjects, len(els)
+
+
+def get_visible_text(soup):
+    # Additional check for pages with JavaScript redirects
+    if 'location.replace(' in soup.getText():
+        result = re.search('else\n {4}location.replace[(]"(.*)"[)];', soup.getText())
```

Comment:
huh, that's some kick-ass regex. What does it do? Would be good to leave a comment in the line above to document it.

Reply:
Added a comment to explain the regex